core/migration_external_user_id.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2022 Board of Trustees of the University of Illinois.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"fmt"
+	"log"
+
+	"go.mongodb.org/mongo-driver/bson"
```

Comment:
We should not introduce mongo dependencies in the `core` package. Any code that needs mongo dependencies should go in the `storage` package (in driven/storage).